godel_simple_gui/src/states/simulating_state.cpp

```diff
@@ -23,7 +19,7 @@ void godel_simple_gui::SimulatingState::onStart(BlendingWidget& gui)
 
   sim_client_ =
       gui.nodeHandle().serviceClient<godel_msgs::SelectMotionPlan>(SELECT_MOTION_PLAN_SERVICE);
-  QtConcurrent::run(this, &SimulatingState::simulateAll);
+  QtConcurrent::run(this, &SimulatingState::simulateAll, boost::ref(gui));
 }
```

Comment:
the action server is contained in the gui object as a public function, therefore we need to pass the object to use it.  This is the approach of the simple gui.